src/micro_component.rs

```diff
@@ -0,0 +1,126 @@
+use gtk::glib::{self, Sender};
+
+use std::cell::Ref;
+use std::cell::RefCell;
+use std::cell::RefMut;
+use std::rc::Rc;
+
+pub struct MicroComponent<Model: MicroModel> {
+    model: Option<Rc<RefCell<Model>>>,
+    widgets: Option<Rc<RefCell<Model::Widgets>>>,
```

Comment:
Looks good overall! I don't think those can be optional though.

Reply:
Fixed